pytensor/tensor/rewriting/elemwise.py

```diff
@@ -66,9 +67,12 @@ def print_profile(cls, stream, prof, level=0):
                 print(blanc, n, ndim[n], file=stream)
 
     def candidate_input_idxs(self, node):
-        if isinstance(node.op.scalar_op, aes.Composite) and len(node.outputs) > 1:
-            # TODO: Implement specialized InplaceCompositeOptimizer with logic
-            #  needed to correctly assign inplace for multi-output Composites
+        # TODO: Implement specialized InplaceCompositeOptimizer with logic
+        #  needed to correctly assign inplace for multi-output Composites
+        #  and ScalarLoops
+        if isinstance(node.op.scalar_op, ScalarLoop):
+            return []
+        if isinstance(node.op.scalar_op, aes.Composite) and (len(node.outputs) > 1):
```

Comment:
`Composite` subclasses `ScalarLoop` now, so why  `Composite` in the 2nd check?

Reply:
Composite subclasses `ScalarOpHasInnerGraph` not `ScalarLoop`